### PR TITLE
CommandPalette: Design tweaks and design fixes 

### PR DESCRIPTION
**Pull request by torkelo**
Trying to explore how we can improve command palette design so it can take the role of global search 

* [x] Increase max width a bit
* [x] Increase max height a bit 
* [x] Change background to primary to match our other modals and popovers 
* [x] Tweaks to item hover / active style to work with the new primary background 
* [x] Make headings medium font-weight 
* [x] Fix inconsistent spacing for the first heading (missing top padding) 
* [x] Change placeholder from "Search Grafana"  to "Search or jump to..." 
* [x] Fix some padding / alignment issues 
* [x] Simplify styles a bit 

TODO (in this or follow-up PRs)
* [ ] Need to visually (through indent or other) separate headings I feel (they are almost visually indistinguishable from items)  
   * [x] Added small horizontal padding to items to make item border-radius visible, also adds a little bit of indent compared to headings. One thing we could do is also add icons to items (but not headings), then it would be more clear. Icons could be based on type (dashboard, page, action could have bolt icon?) 
* [ ] Really Really want page and all navigation actions to be proper links that show links on hover and support CTRL+click, this is a bit of a showstopper for me if we are to use kbar as global search. 
* [ ] Add support showing folder for items with folder 
* [ ] Improve mobile responsive view (The position of the modal is not great on mobile, padding space at the top) 
* [ ] Improve position so that it feels more natural when opened by clicking search bar (related to https://github.com/grafana/grafana/pull/61667) 

Current state: 

Screenshot from 2023-01-24 13-46-41

### CODE REVIEW

**Comment by ashharrison90** [comment]
@torkelo i have a draft here which tackles some of the bullet points you mention: https://github.com/grafana/grafana/pull/61991

> Really Really want page and all navigation actions to be proper links that show links on hover and support CTRL+click, this is a bit of a showstopper for me if we are to use kbar as global search.

agreed this one is important. could get a little messy though as i don't think there's any natural way to do this in `kbar`. let's try and do this in a standalone PR away from these styling tweaks? 🤔

**Comment by torkelo** [comment]
@ashharrison90 yea, I agree we can look at the links issue separately, but if we can't solve that we should abandon kbar.

**Comment by ashharrison90** [comment]
@torkelo i have no idea 😄 the main problem is deciding which bits to use from each 😅 

my 2c: 
- like the size changes in this
- like the color changes in this, especially the difference in color between the input and results
  - not in love with the hover states 🤔
- like the section heading style changes
  - i made pretty much the same change. worth mentioning that when we were mobbing others thought the font size was too small which i still disagree with 😅 
  
things from the other PR i think we need:
- loading spinner when dashboard search is in progress
- positioning above the search bar
- removing recent dashboards when searching
- removing the search command palette action when the `topNavCommandPalette` toggle is enabled

but it's subjective 🤷

**Comment by torkelo** [comment]
@ashharrison90 
> not in love with the hover states

Me neither but it's pretty close to our other hover states. I would prefer blue link color + underline, but since it's not real links :shrug:

**Comment by ashharrison90** [comment]
@torkelo ah sorry ignore the comment about the hover state. think i switched branch without restarting the server and it didn't quite rebuild the css properly. think it actually looks good now 👌 

will discuss in squad sync now the best way to deliver things. suspect merging one into main first is probably easier 👍

**Comment by amy-super** [comment]
I'll leave you all to figure out what to merge where, but I'm onboard with the styling changes. I would strongly prefer that we not use blue underline link styling as it will be quite overwhelming when users are faced with a list of results.

File: public/locales/en-US/grafana.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
       "search": "Search"
     },
     "search-box": {
-      "placeholder": "Search Grafana"
+      "placeholder": "Search or jump to..."
```
Comment (ashharrison90) [inline public/locales/en-US/grafana.json:16]: do you think we'll have any issues with this being different to the top search bar placeholder now, or should we try to keep them the same? 🤔

**Review by ashharrison90 — COMMENTED** [review]
minor nits/questions.

i'll wait for @joshhunt to take a look as well 👍

**Comment by joshhunt** [comment]
@torkelo @ashharrison90 I am about to raise PR to fix the feature flag specifically, then i can take a peek at these changes.

File: public/app/features/commandPalette/CommandPalette.tsx [hunk 2]
```diff
@@ -129,47 +127,39 @@ const getSearchStyles = (theme: GrafanaTheme2) => ({
     },
   }),
   animator: css({
-    maxWidth: theme.breakpoints.values.sm, // supposed to be 600...
+    maxWidth: theme.breakpoints.values.lg,
```
[the remark this answers is not in the document]
Comment (amy-super) [reply public/app/features/commandPalette/CommandPalette.tsx:130]: I like the `md` version better personally.

File: public/locales/en-US/grafana.json (hunk 1 above)
Comment (torkelo) [reply public/locales/en-US/grafana.json:16]: @ashharrison90 I was thinking we change the placeholder text of global search input to match

File: public/app/features/commandPalette/CommandPalette.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/features/commandPalette/CommandPalette.tsx:130]: Agree for the medium width. I think the large just the modal too wide and mostly empty.

File: public/app/features/commandPalette/CommandPalette.tsx [hunk 3]
```diff
@@ -129,47 +127,39 @@ const getSearchStyles = (theme: GrafanaTheme2) => ({
...
     border: 'none',
-    background: theme.colors.background.canvas,
-    color: theme.colors.text.primary,
-    borderBottom: `1px solid ${theme.colors.border.medium}`,
+    background: theme.components.input.background,
+    color: theme.components.input.text,
+    borderBottom: `1px solid ${theme.colors.border.weak}`,
+  }),
+  resultsContainer: css({
+    paddingBottom: theme.spacing(1),
```
Comment (joshhunt) [inline public/app/features/commandPalette/CommandPalette.tsx:151]: I don't think this is needed (or looks good). I think it's weird that items are clipped before the bottom of the modal 

image

File: public/app/features/commandPalette/CommandPalette.tsx [hunk 4]
```diff
@@ -129,47 +127,39 @@ const getSearchStyles = (theme: GrafanaTheme2) => ({
...
   }),
-
   // Virtual list measures margin incorrectly, so we need to split padding before/after border
   // over and inner and outer element
```
Comment (joshhunt) [inline public/app/features/commandPalette/CommandPalette.tsx:154]: As the code is now, this comment is no longer accurate and should be removed.

However, I think that section headers need additional 8px above them to separate them from the previous section and prevent the hover background from being flush against the border:

image

Additionally, I think that there should be ~8px _less_ padding _under_ section headers - i think the section header is too far away from it's items.

File: public/app/features/commandPalette/ResultItem.tsx [hunk 5]
```diff
@@ -63,31 +62,33 @@ export const ResultItem = React.forwardRef(
...
+    activeRow: css({
+      color: theme.colors.text.maxContrast,
+      background: theme.colors.emphasize(theme.colors.background.primary, 0.03),
       '&:before': {
-        display: isActive ? 'block' : 'none',
+        display: 'block',
         content: '" "',
         position: 'absolute',
-        left: 0,
+        left: 4,
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/features/commandPalette/ResultItem.tsx:87]: more noticable to me on light theme - looks like a bug

image

**Review by joshhunt — COMMENTED** [review]
Just left my two cents ✨ 

I think this looks more Grafana-y, so positive on this overall! 👍🏻 

I spent a lot of time getting confused over padding before/after section headers when i tweaked them last week...

Personally I think the section headers are too small at this font size with this dark of a grey. The contrast just isn't there for me.

**Comment by joshhunt** [comment]
Remaining todos seem like great things to tackle in separate PRs :)

File: public/app/features/commandPalette/CommandPalette.tsx (hunk 3 above)
Comment (torkelo) [reply public/app/features/commandPalette/CommandPalette.tsx:151]: @joshhunt yea, not sure what is best, but this was to replicate the existing styles that added padding to the last item (With a comment that is was for the above reason). But to add it to the last item required complicated logic, so found this easier. 

And I think this is more correct. Like if you have scroll inside a Modal body I would not expect the content to clip at the edge of the modal, the modal has a padding and the content should never be able to go outside the content padding.  

But don't have any strong feelings here, was mainly to replicate the existing last item padding with simpler logic.

File: public/app/features/commandPalette/ResultItem.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/commandPalette/ResultItem.tsx:87]: @joshhunt this fix a bug that it was at the edge of the modal, with zero padding (unlike our other modals) and unlike github's search

Maybe 4 is too much and it should be 2?

image

[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/commandPalette/ResultItem.tsx:87]: ah, stupid me I see the bug your talking about now.

File: public/app/features/commandPalette/CommandPalette.tsx (hunk 4 above)
Comment (torkelo) [reply public/app/features/commandPalette/CommandPalette.tsx:154]: @joshhunt oh, nicely spotted on the headings & margins, added a top margin to section headings (but not the first)

Regarding section heading spacing, fixed that as well (was only 4px off, not 8, if I make it buttom padding 8 it look very strange when the item under the header has hover/active state) 

image

Comment (torkelo) [reply public/app/features/commandPalette/CommandPalette.tsx]: the above screenshot shows the current commited section header padding,  theme.spacing(1.5, 2, 1.5, 2),
if I change the bottom padding to just 1 (8px) it looks like this:
image

**Comment by torkelo** [comment]
@amy-super 

> I would strongly prefer that we not use blue underline link styling as it will be quite overwhelming when users are faced with a list of results.

I was only suggesting that for the hover state state, which I don't think is overwhelming. We use it for dashboard links in dashboad list panel. Think it's a nice way to signify something is a link that will take you to another view/page 

image

File: public/app/features/commandPalette/CommandPalette.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/commandPalette/CommandPalette.tsx]: I think it depends on how we add folders and tags. There are really long folder and dashboard names, and dashboards with many tags. I think an md modal will get really ugly with wrapping or cut text and tags. 

But I can leave it md for now. but pretty sure we need to extend it if we want this to take the role of global search

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

File: public/app/features/commandPalette/CommandPalette.tsx (hunk 3 above)
Comment (joshhunt) [reply public/app/features/commandPalette/CommandPalette.tsx:151]: Pages don't have 8px padding at the bottom of the window that clips before the edge of the screen.

Comment (torkelo) [reply public/app/features/commandPalette/CommandPalette.tsx:151]: @joshhunt that's a good argument, I am fine with removing it. 👍

**Pull request merged**